Lib/urllib/parse.py

```diff
@@ -414,6 +417,11 @@ def _checknetloc(netloc):
             raise ValueError("netloc '" + netloc + "' contains invalid " +
                              "characters under NFKC normalization")
 
+def _remove_unsafe_bytes_from_url(url):
```

Comment:
A refactor of of this code was required in Python 3.8 and below as we special case parsing for http urls.
Please review if this refactor is alright, @gpshead  :)